samples/net/wifi fails to build on cc3220sf_launchxl

```
In file included from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:51:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:21:2: error: redeclaration of enumerator 'Reset_IRQn'
  Reset_IRQn                    = -15,
  ^~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/include/arch/arm/cortex_m/cmsis.h:17,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/cortex_m/stack.h:27,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/kernel_arch_data.h:53,
                 from /home/apboie/projects/zephyr2/zephyr/kernel/include/kernel_structs.h:68,
                 from /home/apboie/projects/zephyr2/zephyr/include/sys/mutex.h:113,
                 from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:19:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:21:2: note: previous definition of 'Reset_IRQn' was here
  Reset_IRQn                    = -15,
  ^~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:51:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:22:2: error: redeclaration of enumerator 'NonMaskableInt_IRQn'
  NonMaskableInt_IRQn           = -14,
  ^~~~~~~~~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/include/arch/arm/cortex_m/cmsis.h:17,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/cortex_m/stack.h:27,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/kernel_arch_data.h:53,
                 from /home/apboie/projects/zephyr2/zephyr/kernel/include/kernel_structs.h:68,
                 from /home/apboie/projects/zephyr2/zephyr/include/sys/mutex.h:113,
                 from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:19:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:22:2: note: previous definition of 'NonMaskableInt_IRQn' was here
  NonMaskableInt_IRQn           = -14,
  ^~~~~~~~~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:51:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:23:2: error: redeclaration of enumerator 'HardFault_IRQn'
  HardFault_IRQn                = -13,
  ^~~~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/include/arch/arm/cortex_m/cmsis.h:17,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/cortex_m/stack.h:27,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/kernel_arch_data.h:53,
                 from /home/apboie/projects/zephyr2/zephyr/kernel/include/kernel_structs.h:68,
                 from /home/apboie/projects/zephyr2/zephyr/include/sys/mutex.h:113,
                 from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:19:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:23:2: note: previous definition of 'HardFault_IRQn' was here
  HardFault_IRQn                = -13,
  ^~~~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:51:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:24:2: error: redeclaration of enumerator 'MemoryManagement_IRQn'
  MemoryManagement_IRQn         = -12,
  ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/include/arch/arm/cortex_m/cmsis.h:17,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/cortex_m/stack.h:27,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/kernel_arch_data.h:53,
                 from /home/apboie/projects/zephyr2/zephyr/kernel/include/kernel_structs.h:68,
                 from /home/apboie/projects/zephyr2/zephyr/include/sys/mutex.h:113,
                 from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:19:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:24:2: note: previous definition of 'MemoryManagement_IRQn' was here
  MemoryManagement_IRQn         = -12,
  ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:51:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:25:2: error: redeclaration of enumerator 'BusFault_IRQn'
  BusFault_IRQn                 = -11,
  ^~~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/include/arch/arm/cortex_m/cmsis.h:17,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/cortex_m/stack.h:27,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/kernel_arch_data.h:53,
                 from /home/apboie/projects/zephyr2/zephyr/kernel/include/kernel_structs.h:68,
                 from /home/apboie/projects/zephyr2/zephyr/include/sys/mutex.h:113,
                 from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:19:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:25:2: note: previous definition of 'BusFault_IRQn' was here
  BusFault_IRQn                 = -11,
  ^~~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:51:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:26:2: error: redeclaration of enumerator 'UsageFault_IRQn'
  UsageFault_IRQn               = -10,
  ^~~~~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/include/arch/arm/cortex_m/cmsis.h:17,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/cortex_m/stack.h:27,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/kernel_arch_data.h:53,
                 from /home/apboie/projects/zephyr2/zephyr/kernel/include/kernel_structs.h:68,
                 from /home/apboie/projects/zephyr2/zephyr/include/sys/mutex.h:113,
                 from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:19:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:26:2: note: previous definition of 'UsageFault_IRQn' was here
  UsageFault_IRQn               = -10,
  ^~~~~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:51:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:27:2: error: redeclaration of enumerator 'SVCall_IRQn'
  SVCall_IRQn                   =  -5,
  ^~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/include/arch/arm/cortex_m/cmsis.h:17,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/cortex_m/stack.h:27,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/kernel_arch_data.h:53,
                 from /home/apboie/projects/zephyr2/zephyr/kernel/include/kernel_structs.h:68,
                 from /home/apboie/projects/zephyr2/zephyr/include/sys/mutex.h:113,
                 from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:19:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:27:2: note: previous definition of 'SVCall_IRQn' was here
  SVCall_IRQn                   =  -5,
  ^~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:51:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:28:2: error: redeclaration of enumerator 'DebugMonitor_IRQn'
  DebugMonitor_IRQn             =  -4,
  ^~~~~~~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/include/arch/arm/cortex_m/cmsis.h:17,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/cortex_m/stack.h:27,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/kernel_arch_data.h:53,
                 from /home/apboie/projects/zephyr2/zephyr/kernel/include/kernel_structs.h:68,
                 from /home/apboie/projects/zephyr2/zephyr/include/sys/mutex.h:113,
                 from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:19:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:28:2: note: previous definition of 'DebugMonitor_IRQn' was here
  DebugMonitor_IRQn             =  -4,
  ^~~~~~~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:51:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:29:2: error: redeclaration of enumerator 'PendSV_IRQn'
  PendSV_IRQn                   =  -2,
  ^~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/include/arch/arm/cortex_m/cmsis.h:17,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/cortex_m/stack.h:27,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/kernel_arch_data.h:53,
                 from /home/apboie/projects/zephyr2/zephyr/kernel/include/kernel_structs.h:68,
                 from /home/apboie/projects/zephyr2/zephyr/include/sys/mutex.h:113,
                 from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:19:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:29:2: note: previous definition of 'PendSV_IRQn' was here
  PendSV_IRQn                   =  -2,
  ^~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:51:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:30:2: error: redeclaration of enumerator 'SysTick_IRQn'
  SysTick_IRQn                  =  -1,
  ^~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/include/arch/arm/cortex_m/cmsis.h:17,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/cortex_m/stack.h:27,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/kernel_arch_data.h:53,
                 from /home/apboie/projects/zephyr2/zephyr/kernel/include/kernel_structs.h:68,
                 from /home/apboie/projects/zephyr2/zephyr/include/sys/mutex.h:113,
                 from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:19:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:30:2: note: previous definition of 'SysTick_IRQn' was here
  SysTick_IRQn                  =  -1,
  ^~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:51:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:31:3: error: conflicting types for 'CMSIS_IRQn_Type'
 } CMSIS_IRQn_Type;
   ^~~~~~~~~~~~~~~
In file included from /home/apboie/projects/zephyr2/zephyr/include/arch/arm/cortex_m/cmsis.h:17,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/cortex_m/stack.h:27,
                 from /home/apboie/projects/zephyr2/zephyr/arch/arm/include/kernel_arch_data.h:53,
                 from /home/apboie/projects/zephyr2/zephyr/kernel/include/kernel_structs.h:68,
                 from /home/apboie/projects/zephyr2/zephyr/include/sys/mutex.h:113,
                 from /home/apboie/projects/zephyr2/zephyr/lib/libc/newlib/libc-hooks.c:19:
/home/apboie/projects/zephyr2/zephyr/soc/arm/ti_simplelink/cc32xx/soc.h:31:3: note: previous declaration of 'CMSIS_IRQn_Type' was here
 } CMSIS_IRQn_Type;
   ^~~~~~~~~~~~~~~
zephyr/lib/libc/newlib/CMakeFiles/lib__libc__newlib.dir/build.make:62: recipe for target 'zephyr/lib/libc/newlib/CMakeFiles/lib__libc__newlib.dir/libc-hooks.c.obj' failed
make[2]: *** [zephyr/lib/libc/newlib/CMakeFiles/lib__libc__newlib.dir/libc-hooks.c.obj] Error 1
CMakeFiles/Makefile2:1050: recipe for target 'zephyr/lib/libc/newlib/CMakeFiles/lib__libc__newlib.dir/all' failed
make[1]: *** [zephyr/lib/libc/newlib/CMakeFiles/lib__libc__newlib.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

--- a/soc/arm/ti_simplelink/cc32xx/soc.h
+++ b/soc/arm/ti_simplelink/cc32xx/soc.h
@@ -4,6 +4,9 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#ifndef TI_SIMPLELINK_CC32XX_SOC_H_
+#define TI_SIMPLELINK_CC32XX_SOC_H_
+
 #include <inc/hw_types.h>
 #include <driverlib/prcm.h>
 
@@ -34,3 +37,6 @@ typedef enum {
 #define __MPU_PRESENT                  0 /* Zephyr has no MPU support */
 #define __NVIC_PRIO_BITS               DT_NUM_IRQ_PRIO_BITS
 #define __Vendor_SysTickConfig         0 /* Default to standard SysTick */
+
+#endif /* TI_SIMPLELINK_CC32XX_SOC_H_ */
+

